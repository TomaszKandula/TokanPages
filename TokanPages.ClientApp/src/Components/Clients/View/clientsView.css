.clients-render-images {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.clients-caption {
    text-align: center;
    font-size: 2rem !important;
    font-weight: 500 !important;
    color: #6367ef;
}

.clients-section {
    background-color: white;
    padding-top: 5px;
    padding-bottom: 5px;
}

@media (max-width: 900px) {
    .clients-section {
        padding-top: 50px;
        padding-bottom: 50px;
    }
}

.clients-logo {
    padding-left: 5%;
    padding-right: 5%;
}

@media (max-width: 900px) {
    .clients-logo {
        padding-left: 2%;
        padding-right: 2%;
        margin: 25px;
    }
}
