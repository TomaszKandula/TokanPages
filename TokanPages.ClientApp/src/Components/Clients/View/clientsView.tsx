import * as React from "react";
import { useSelector } from "react-redux";
import { ApplicationState } from "../../../Store/Configuration";
import { GET_ICONS_URL } from "../../../Api";
import { ClientImageDto, ClientsContentDto } from "../../../Api/Models";
import { Skeleton } from "../../../Shared/Components";
import { useDimensions } from "../../../Shared/Hooks";
import { v4 as uuidv4 } from "uuid";
import Validate from "validate.js";

interface ClientsViewProps {
    className?: string;
}

interface ClientsViewExtendedProps extends ClientsContentDto {
    isLoading: boolean;
    isDesktop: boolean;
}

const RenderCaption = (props: ClientsViewExtendedProps): React.ReactElement | null => {
    if (!Validate.isEmpty(props?.caption)) {
        return (
            <Skeleton isLoading={props.isLoading} mode="Text" height={40}>
                <p className="is-size-3	has-text-centered has-text-link">{props?.caption?.toUpperCase()}</p>
            </Skeleton>
        );
    }

    return null;
};

const RenderImages = (props: ClientsViewExtendedProps): React.ReactElement => {
    const getImagePath = (value: string): string => `${GET_ICONS_URL}/${value}`;

    return (
        <div className="is-flex is-flex-wrap-wrap is-justify-content-center is-align-items-center">
            {props?.images?.map((item: ClientImageDto, _index: number) => (
                <div className={`${props.isDesktop ? "mx-6" : "m-6"}`} key={uuidv4()}>
                    <Skeleton isLoading={props.isLoading} mode="Rect" width={100} height={100}>
                        <img
                            src={getImagePath(item.path)}
                            loading="lazy"
                            alt={`An image of ${item.name}`}
                            title="Clients"
                            height={item.heigh}
                            width={item.width}
                            className="lazyloaded"
                        />
                    </Skeleton>
                </div>
            ))}
        </div>
    );
};

export const ClientsView = (props: ClientsViewProps): React.ReactElement => {
    const media = useDimensions();
    const data = useSelector((state: ApplicationState) => state.contentPageData);
    const clients = data?.components.sectionClients;

    return (
        <>
            <section className={props.className}>
                <div className="bulma-container">
                    <div className={`${media.isDesktop ? "py-6" : "py-4"}`}>
                        <RenderCaption isLoading={data?.isLoading} isDesktop={media.isDesktop} {...clients} />
                        <RenderImages isLoading={data?.isLoading} isDesktop={media.isDesktop} {...clients} />
                    </div>
                </div>
            </section>
        </>
    );
};
