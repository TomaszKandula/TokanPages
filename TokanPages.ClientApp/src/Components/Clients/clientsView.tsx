import * as React from "react";
import Skeleton from "@material-ui/lab/Skeleton";
import { Box, Container, Typography } from "@material-ui/core";
import { IGetClientsContent } from "../../Redux/States/Content/getClientsContentState";
import { ICONS_PATH } from "../../Shared/constants";
import ClientsStyle from "./clientsStyle";
import Validate from "validate.js";

const ClientsView = (props: IGetClientsContent): JSX.Element => 
{
    const classes = ClientsStyle();

    const getImagePath = (value: string): string => 
    {
        return `${ICONS_PATH}${value}`;
    }

    const RenderCaption = (): JSX.Element | null => 
    {
        if (!Validate.isEmpty(props.content?.caption))
        {
            return(
                <Typography className={classes.caption}>
                    {props.content?.caption?.toUpperCase()}                    
                </Typography>
            );
        }

        return null;
    }

    const RenderImages = (): JSX.Element => 
    {
        return(
            <Box pt={4} display="flex" flexWrap="wrap" justifyContent="center">
                {props.content?.images.map((item, index) => (
                    <img key={index} src={getImagePath(item)} alt="" className={classes.logo} />
                ))}
            </Box>
        );
    }

    return(
        <>
            <div className={classes.divider}></div>
            <section className={classes.section}>
                <Container maxWidth="lg">
                    <Box mb={8}>
                        {props.isLoading ? <Skeleton variant="text" /> : <RenderCaption />}
                    </Box>
                    {props.isLoading ? <Skeleton variant="rect" height="48px" /> : <RenderImages />}
                </Container>
            </section>
        </>
    );
}

export default ClientsView;
