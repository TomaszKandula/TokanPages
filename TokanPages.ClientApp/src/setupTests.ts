// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
