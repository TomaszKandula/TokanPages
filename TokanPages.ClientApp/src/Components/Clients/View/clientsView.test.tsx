import "../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { ClientsView } from "./clientsView";

describe("test component: clientsView", () => {
    it("should render correctly '<ClientsView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <ClientsView background="class-colour-white" />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
