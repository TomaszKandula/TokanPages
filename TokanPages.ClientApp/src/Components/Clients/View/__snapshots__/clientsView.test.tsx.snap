// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: clientsView should render correctly '<ClientsView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="clients-section class-colour-white"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="clients-render-images mt-15 mb-64"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="clients-section class-colour-white"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="clients-render-images mt-15 mb-64"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
